File: Java/NetBeans/DSNV/src/dsnv/DSNV.java
package dsnv;

public class DSNV {

    NV[] ds;
    int soNV;

    public DSNV(int soNVMax) {
        ds = new NV[soNVMax];
        soNV = 0;
    }

    public void them(NV nv) {
        if (soNV < ds.length) {
            ds[soNV++] = nv;
        }
    }

    public void lietKe() {
        for (int i = 0; i < soNV; i++) {
            ds[i].hienThi();
        }
    }

    public void lietKe(String loaiNV) {
        for (int i = 0; i < soNV; i++) {
            if (ds[i].loaiNV().equalsIgnoreCase(loaiNV)) {
                ds[i].hienThi();
            }
        }
    }

    public double tongLuong() {
        double S = 0;
        for (int i = 0; i < soNV; i++) {
            S += ds[i].layLuong();
        }
        return S;
    }

    public void dsHDDH() {
        for (int i = 0; i < soNV; i++) {
            if (ds[i].laNVHDDH()) {
                ds[i].hienThi();
            }
        }
    }
}

File: C#/QLNV/QLNV/DSNV.cs
namespace QLNV
{
    internal class DSNV
    {
        private List<NV> ds;

        public DSNV()
        {
            ds = new List<NV>();
        }

        public bool them(NV nv)
        {
            if (!ds.Contains(nv))
            {
                ds.Add(nv);
                return true;
            }

            return false;
        }

        public void lietKe()
        {
            foreach (NV nv in ds)
            {
                nv.hienThi();
            }
        }

        public void lietKe(string loaiNV)
        {
            foreach (NV nv in ds)
            {
                if (nv.loaiNV().Equals(loaiNV))
                {
                    nv.hienThi();
                }
            }
        }

        public double tongLuong()
        {
            double tong = 0;

            foreach (NV nv in ds)
            {
                tong += nv.layLuong();
            }

            return tong;
        }

        public void dsHDDH()
        {
            foreach (NV nv in ds)
            {
                if (nv.laNVHDDH())
                {
                    nv.hienThi();
                }
            }
        }
    }
}
